problem_builder/public/js/lms_util.js

```diff
@@ -0,0 +1,120 @@
+        this.show = function () {
+            if (typeof video.resizer === 'undefined'){
+                // This one is tricky: but it looks like resizer is undefined only if the video is on the
+                // step that is initially visible (and then no resizing is necessary)
+                return;
+            }
+            video.resizer.align();
+        };
+        /**
+         * Videos should be paused when user leaves, a step containing the video. There is was a proposed implementation
+         * but since it didn't work on every system we decided to drop it (it was out of scope for current task
+         * nevertheless). See OC-1441 for details.
+         */
+        this.hide = function () {};
+    }
+    
+    /**
+     *  Manager for Plot Xblocks. Handles updating a plot before displaying it.
+     * @param plot
+     */
+    function PlotManager(plot) {
+        this.show = function () {
+            plot.update();
+        };
+        this.hide = function () {};
+    }
+
+
+    function ChildManager(xblock_element, runtime) {
+
+        var Managers = {
+            'video': VideoManager,
+            'sb-plot': PlotManager
+        };
+
+        var children = runtime.children(xblock_element);
+
+        /**
+         * A list of managers for children than need special care when showing or hiding.
+         *
+         * @type {show, hide}[]
+         */
+        var managedChildren = [];
+
+        /***
+         * This is a workaround for issue where jquery.xblock.Runtime doesn't return HTML Xblocks when querying
+         * for children.
+         *
+         * This can be removed when:
+         *
+         * * We allow inclusion of Ooyala blocks inside ProblemBuilder and our clients migrate to Ooyala, in this case
+         *   we may drop special handling of HTML blocks. See discussions in OC-1441.
+         * * We include HTML blocks in runtime.children for runtime of jquery.xblock, then just add
+         *   `html: HtmlManager` to `Managers`, and remove this block.
+         */
+        $("div.xblock.xblock-student_view.xmodule_HtmlModule", xblock_element).each(function (idx, element) {
+            managedChildren.push(new HtmlManager({element:element}));
+        });
+
+        for (var idx = 0; idx < children.length; idx++){
+            var child = children[idx];
+            // NOTE: While following assertion is true for e.g Video blocks:
+            // child.type == $(child.element).data('block-type') it is invalidated by for all sb-* blocks
+            var type =  $(child.element).data('block-type');
+            var constructor = Managers[type];
+            if (typeof constructor === 'undefined'){
+                // This block does not requires special care, moving on 
+                continue ;
+            }
+            managedChildren.push(new constructor(child));
+        }
+        
+        this.show = function () {
+            for (var idx = 0; idx<managedChildren.length; idx++){
```

Comment:
@jbzdak Can you try and make the spacing in these `for` loops consistent throughout this file? The following version

```js
(var idx = 0; idx < managedChildren.length; idx++)
```

is best for readability I think.